variables.tf

```diff
@@ -2372,3 +2372,207 @@ variable "create_egress_only_igw" {
   type        = bool
   default     = true
 }
+
+variable "s3_endpoint_policy" {
+  description = "Custom IAM policy for S3 VPC endpoint"
+  type        = string
+  default     = ""
+}
+
+variable "dynamodb_endpoint_policy" {
+  description = "Custom IAM policy for DynamoDB VPC endpoint"
+  type        = string
+  default     = ""
+}
+
+variable "codebuild_endpoint_policy" {
+  description = "Custom IAM policy for Codebuild VPC endpoint"
+  type        = string
+  default     = ""
+}
+
+variable "codecommit_endpoint_policy" {
+  description = "Custom IAM policy for Code Commit VPC endpoint"
+  type        = string
+  default     = ""
+}
+
+variable "sqs_endpoint_policy" {
+  description = "Custom IAM policy for SQS VPC endpoint"
+  type        = string
+  default     = ""
+}
+
+variable "secretsmanager_endpoint_policy" {
+  description = "Custom IAM policy for Secrets Manager VPC endpoint"
+  type        = string
+  default     = ""
+}
+
+variable "ec2_endpoint_policy" {
+  description = "Custom IAM policy for EC2 VPC endpoint"
+  type        = string
+  default     = ""
+}
+
+variable "ec2_autoscaling_endpoint_policy" {
+  description = "Custom IAM policy for EC2 Autoscaling VPC endpoint"
+  type        = string
+  default     = ""
+}
+
+variable "ecr_api_endpoint_policy" {
+  description = "Custom IAM policy for ECR API VPC endpoint"
+  type        = string
+  default     = ""
+}
+
+variable "ecr_dkr_endpoint_policy" {
```

Comment:
likewise, I see ECR supported but I don't know if that includes the Docker registry API endpoint https://docs.aws.amazon.com/vpc/latest/userguide/integrated-services-vpce-list.html